src/pages/home/report/ContextMenu/BaseReportActionContextMenu.tsx

```diff
@@ -0,0 +1,243 @@
+import lodashIsEqual from 'lodash/isEqual';
+import type {MutableRefObject, RefObject} from 'react';
+import React, {memo, useMemo, useRef, useState} from 'react';
+import {InteractionManager, View} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {ContextMenuItemHandle} from '@components/ContextMenuItem';
+import ContextMenuItem from '@components/ContextMenuItem';
+import useArrowKeyFocusManager from '@hooks/useArrowKeyFocusManager';
+import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import * as Session from '@userActions/Session';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Beta, ReportAction, ReportActions} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {ContextMenuActionPayload} from './ContextMenuActions';
+import ContextMenuActions from './ContextMenuActions';
+import type {ContextMenuType} from './ReportActionContextMenu';
+import {hideContextMenu} from './ReportActionContextMenu';
+
+type BaseReportActionContextMenuOnyxProps = {
+    /** Beta features list */
+    betas: OnyxEntry<Beta[]>;
+
+    /** All of the actions of the report */
+    reportActions: OnyxEntry<ReportActions>;
+};
+
+type BaseReportActionContextMenuProps = BaseReportActionContextMenuOnyxProps & {
+    /** The ID of the report this report action is attached to. */
+    reportID: string;
+
+    /** The ID of the report action this context menu is attached to. */
+    reportActionID: string;
+
+    /** The ID of the original report from which the given reportAction is first created. */
+    // originalReportID is used in withOnyx to get the reportActions for the original report
+    // eslint-disable-next-line react/no-unused-prop-types
+    originalReportID: string;
+
+    /** If true, this component will be a small, row-oriented menu that displays icons but not text.
+  If false, this component will be a larger, column-oriented menu that displays icons alongside text in each row. */
```

Comment:
Small nitpick on style. Surprised that the linter didn't get that.

```suggestion
    /**
     * If true, this component will be a small, row-oriented menu that displays icons but not text.
     * If false, this component will be a larger, column-oriented menu that displays icons alongside text in each row.
     */
```